ui/components/ui/unit-input/unit-input.component.js

```diff
@@ -117,14 +158,22 @@ export default class UnitInput extends PureComponent {
               onChange={this.handleChange}
               onBlur={this.handleInputBlur}
               onFocus={this.handleInputFocus}
+              onKeyPress={(e) =>
+                this.props.keyPressRegex &&
+                !this.props.keyPressRegex.test(e.key) &&
+                e.preventDefault()
+              }
```

Comment:
Is this necessary if the input is already typed as "number"?

Reply:
probably need to set `min` as well to prevent negative numbers

Reply:
We need to filter beyond just the numbers. The regex catches negative numbers as well.

Great call on `min`. Also added regex filtering `onPaste`